### PR TITLE
Pass build extra options from the upstream to the downstream pipelines

### PR DESCRIPTION
**Pull request by vsebe**
This allows users to override options on demand.
Update Jenkins files to:
- check for EXTRA_*_OPTIONS build parameters
- update Pipeline-Build-Test-All to pass EXTRA_*_OPTIONS to the downstream jobs

[ci skip]

Signed-off-by: Violeta Sebe <vsebe@ca.ibm.com>

### CODE REVIEW

**Comment by vsebe** [comment]
@AdamBrousseau, @jdekonin ready for review

File: buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All [hunk 1]
```diff
@@ -156,7 +156,12 @@ timeout(time: 12, unit: 'HOURS') {
                             TEST_NODE = TEST_NODES[SPEC]
                         }
 
-                        builds["build_test_JDK${SDK_VERSION}_${SPEC}"] = { build(REPO, BRANCH, SHAS, OPENJ9_REPO, OPENJ9_BRANCH, OMR_REPO, OMR_BRANCH, SPEC, SDK_VERSION, BUILD_NODE, TEST_NODE) }
+                        def EXTRA_GETSOURCE_OPTIONS = get_value_by_spec(EXTRA_GETSOURCE_OPTIONS_MAP, SDK_VERSION, SPEC)
+                        def EXTRA_CONFIGURE_OPTIONS = get_value_by_spec(EXTRA_CONFIGURE_OPTIONS_MAP, SDK_VERSION, SPEC)
+                        def EXTRA_MAKE_OPTIONS = get_value_by_spec(EXTRA_MAKE_OPTIONS_MAP, SDK_VERSION, SPEC)
+                        def OPENJDK_CLONE_DIR = get_value_by_spec(OPENJDK_CLONE_DIR_MAP, SDK_VERSION, SPEC)
+
+                        builds["build_test_JDK${SDK_VERSION}_${SPEC}"] = { build(REPO, BRANCH, SHAS, OPENJ9_REPO, OPENJ9_BRANCH, OMR_REPO, OMR_BRANCH, SPEC, SDK_VERSION, BUILD_NODE, TEST_NODE, EXTRA_GETSOURCE_OPTIONS, EXTRA_CONFIGURE_OPTIONS, EXTRA_MAKE_OPTIONS, OPENJDK_CLONE_DIR) }
```
Comment (AdamBrousseau) [inline buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All]: Why are we getting the config options in PBTA and passing down to the build job? If the options are only needed for the compile can the Build job just set these variables itself?

Comment (vsebe) [reply buildenv/jenkins/jobs/pipelines/Pipeline-Build-Test-All]: To allow the user to override them or to allow an upstream job to override them.
The Build job sets them. Note the variables-functions.set_build_extra_options() sets the variables (see if statement). They are set to the default values defined in the variables file defaults.yml.

**Comment by pshipton** [comment]
Note there is a merge conflict.

**Comment by vsebe** [comment]
Merged conflict fixed, ready for review, thank you!

**Comment by vsebe** [comment]
@jdekonin, @AdamBrousseau please review

**Comment by jdekonin** [comment]
Can you please break out the extra_getsource_options related to openssl into a separate pr?  While closely related, I believe it complicates this set of changes.

**Comment by vsebe** [comment]
@jdekonin - please see https://github.com/eclipse/openj9/pull/4231. 
PR#4231 should be deliver first.

**Comment by vsebe** [comment]
https://github.com/eclipse/openj9/pull/4231 is merged
@AdamBrousseau review for review

File: buildenv/jenkins/common/pipeline-functions [hunk 2]
```diff
@@ -331,7 +335,7 @@ def workflow(SDK_VERSION, SPEC, SHAS, OPENJDK_REPO, OPENJDK_BRANCH, OPENJ9_REPO,
 
     // compile the source and build the SDK
     def BUILD_JOB_NAME = "Build-JDK${SDK_VERSION}-${SPEC}"
-    jobs["build"] = build(BUILD_JOB_NAME, OPENJDK_REPO, OPENJDK_BRANCH, SHAS['OPENJDK'], OPENJ9_REPO, OPENJ9_BRANCH, SHAS['OPENJ9'], OMR_REPO, OMR_BRANCH, SHAS['OMR'], VARIABLE_FILE, VENDOR_REPO, VENDOR_BRANCH, VENDOR_CREDENTIALS_ID, BUILD_NODE, SETUP_LABEL, BUILD_IDENTIFIER, ghprbGhRepository, ghprbActualCommit, GITHUB_SERVER)
+    jobs["build"] = build(BUILD_JOB_NAME, OPENJDK_REPO, OPENJDK_BRANCH, SHAS['OPENJDK'], OPENJ9_REPO, OPENJ9_BRANCH, SHAS['OPENJ9'], OMR_REPO, OMR_BRANCH, SHAS['OMR'], params.VARIABLE_FILE, params.VENDOR_REPO, params.VENDOR_BRANCH, params.VENDOR_CREDENTIALS_ID, params.BUILD_NODE, SETUP_LABEL, BUILD_IDENTIFIER, ghprbGhRepository, ghprbActualCommit, GITHUB_SERVER, EXTRA_GETSOURCE_OPTIONS, EXTRA_CONFIGURE_OPTIONS, EXTRA_MAKE_OPTIONS, OPENJDK_CLONE_DIR)
```
[the remark this answers is not in the document]
Comment (vsebe) [reply buildenv/jenkins/common/pipeline-functions]: this looks like a merge issue, will remove `params.`

**Comment by AdamBrousseau** [comment]
Do you have links to test builds with and without each of the 4 params set?

**Comment by vsebe** [comment]
@pshipton please review, thank you!

**Comment by vsebe** [comment]
Fixed merge conflict.

**Comment by vsebe** [comment]
@pshipton ready for review

**Pull request merged**